### PR TITLE
Preliminary cut of editing

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: es6/jsxcad-api-shape.js [hunk 1]
```diff
@@ -3318,6 +3318,9 @@ const edit = Shape.chainable(
 
 Shape.registerMethod('edit', edit);
 
+// TODO: Add an option to include a virtual segment at the target of the last
```
Comment (BarbourSmith) [inline es6/jsxcad-api-shape.js:3321]: ❤️

File: es6/jsxcad-api-v1-png.js [hunk 2]
```diff
@@ -129,10 +13,7 @@ const readPng = async (path) => {
   return raster;
 };
 
-const readPngAsContours = async (
-  path,
-  { by = 10, tolerance = 5, to = 256 } = {}
-) => {
+const readPngAsContours = async (path, bands = [128, 256]) => {
```
Comment (BarbourSmith) [inline es6/jsxcad-api-v1-png.js:16]: How does this work? It pulls in all the pixels as points?

**Review by BarbourSmith — APPROVED** [review]
Looks good to me :)

File: es6/jsxcad-api-v1-png.js (hunk 2 above)
Comment (pentacular) [reply es6/jsxcad-api-v1-png.js:16]: The pixels are converted to contour lines based on the grayscale value -- like on a map.

They come in as segments.

**Pull request merged**